pkg/infrastructure/clusterapi/ignition.go

```diff
@@ -0,0 +1,126 @@
+package clusterapi
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	igntypes "github.com/coreos/ignition/v2/config/v3_2/types"
+	"sigs.k8s.io/yaml"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/installer/cmd/openshift-install/command"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/lbconfig"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
+)
+
+const (
+	infrastructureFilepath = "/opt/openshift/manifests/cluster-infrastructure-02-config.yml"
+
+	// replaceable is the string that precedes the encoded data in the ignition data.
+	// The data must be replaced before decoding the string, and the string must be
+	// prepended to the encoded data.
+	replaceable = "data:text/plain;charset=utf-8;base64,"
+)
+
+// EditIgnition attempts to edit the contents of the bootstrap ignition when the user has selected
+// a custom DNS configuration. Find the public and private load balancer addresses and fill in the
+// infrastructure file within the ignition struct.
+func EditIgnition(_ context.Context, in IgnitionInput, platform string, publicIPAddresses, privateIPAddresses []string) ([]byte, error) {
```

Comment:
Do we need context as an input parameter?

Reply:
We could remove it. It used to be required for some other tasks when we needed to create the bucket (but CAPA does that). It was left here on purpose in the event that we may need it in the future.